lib/content-negotiation.js

```diff
@@ -60,6 +64,13 @@ const mediaTypeMarshallers = {
     },
 };
 
+function resolveDecoder(charset) {
+    if ((charset || "").toLocaleLowerCase("en-US") === "utf-8") {
```

Comment:
why have a special case for utf8 then?

Reply:
because I already have a prepared instance for query strings, it's just a blind optimization here

Reply:
Sorry, but this tends to de-optimize my brain. More noise than benefit IMO.
Also, seems like the no value case (`""`) ends up being wrong, doesn't it?

Reply:
If the charset is falsy (more precisely: undefined), we'll do `return new TextDecoder(undefined)` which is the equivalent of `return new TextDecoder()`.
And the docs say that TextDecoder assumes `UTF-8` by default.

🤔 so that's indeed wrong
not sure what to do if the charset is not specified, though.

Reply:
I'm also interested in what `new TextDecoder("")` does...

Reply:
sure, but that's not what's happening here.

Reply:
the only way I would see this potentially happen is when the content type is `some/content-type;charset=`

Reply:
What do you mean? If charset == "", that's what is happening, isn't it?

Reply:
```js
function parseContentType(contentType) {
    const { type, parameters } = contentTypeParser.parse(contentType);
    return { type, charset: parameters.charset };
}
```
if the charset is not specified, `parameters.charset` will yield `undefined` not `""`

Reply:
gotcha

Reply:
the general question remains: what is the invoker supposed to do if the charset is not explicitly provided? throw?

Reply:
Depends on the contentType I would argue. And some of the cases we can decide (eg spec that the default for text/plain is utf8, per our choosing). But for JSON, my understanding is that utf8 is assumed by default.

Reply:
Fixed in https://github.com/projectriff/node-function-invoker/pull/179/commits/6d550efa727bc0c0a29e65a6208b9dd0257ce4e7.
I went with utf8 as default for `text/plain`. It's also possible to throw now.